### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by machour**
* Added missing packages to composer.json
* Extracted npm assets (should help with #4)
* Updated tests namespace
* Added PHP 7.3 to Travis

Couple of questions in comments

### CODE REVIEW

File: src/CodeFile.php [hunk 1]
```diff
@@ -63,8 +63,6 @@ class CodeFile extends BaseObject
      */
     public function __construct($path, $content, $config = [])
     {
-        parent::__construct($config);
-
```
Comment (machour) [inline src/CodeFile.php]: BaseObject doesn't have a constructor anymore, so this was failing.
Is it okay to still use the constructor here ? (I'm really not understanding the DI refactoring 😅)

File: src/Generator.php [hunk 2]
```diff
@@ -75,9 +76,10 @@
     /**
      * {@inheritdoc}
      */
-    public function init()
+    public function __construct()
     {
-        parent::init();
+
+        //parent::__construct();
```
Comment (machour) [inline src/Generator.php]: I've tried to implement `Initiable`, but the init() method wasn't called.
What's the good way to do things here ?

File: src/Generator.php [hunk 3]
```diff
@@ -311,7 +313,7 @@ public function getTemplatePath()
      */
     public function render($template, $params = [])
     {
-        $view = new View();
+        $view = new View(Yii::getApp(), new Theme());
```
Comment (machour) [inline src/Generator.php]: Is `Yii::getApp()` ok, or should the $app instance be somehow injected?

File: src/generators/extension/default/README.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <?= $generator->title ?>
 
-<?= str_repeat('=', mb_strlen($generator->title, \Yii::getApp()->charset)) ?>
+<?= str_repeat('=', mb_strlen($generator->title, \yii\helpers\Yii::getApp()->encoding)) ?>
```
Comment (machour) [inline src/generators/extension/default/README.md]: What's the shorter way to do this ?

File: src/Generator.php (hunk 2 above)
Comment (hiqsol) [reply src/Generator.php]: Problem with moving everything to constructor is that you just don't have `$this->templates` yet in constructor.
So, templates should be passed to constructor or initialization should be moved to init.

File: src/generators/extension/default/README.md (hunk 4 above)
Comment (hiqsol) [reply src/generators/extension/default/README.md]: `Yii::getEncoding()`

File: src/Generator.php (hunk 3 above)
Comment (hiqsol) [reply src/Generator.php]: Yes, application could be injected in constructor.

File: src/CodeFile.php (hunk 1 above)
Comment (hiqsol) [reply src/CodeFile.php]: In this case you can just remove `$config` from arguments, AFAICS it was not used actually.

File: src/Generator.php (hunk 3 above)
Comment (machour) [reply src/Generator.php]: Just to be sure, using constructor injection like this:
```
public function __construct(\yii\base\Application $app) {
    $this->_app = $app;
}
```

would mean that I have to go through all the `$generator = new XXXGenerator()` in the test cases to add `$this->app` as parameter ? Is there a nicer way to do this ?

Also, I put back the init() method and implemented Initiable, but it never gets called (try putting `die("I'm called")`, nothing will happen). What am I missing here ? 🤔

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**